Tests/configure_and_test_integration_instances.py

```diff
@@ -1110,13 +1151,27 @@ def instance_testing(build: Build, all_module_instances, pre_update):
         logging.info(f'Start of Instance Testing ("Test" button) ({update_status}-update)')
     else:
         logging.info(f'No integrations to configure for the chosen tests. ({update_status}-update)')
-
     for instance in all_module_instances:
         integration_of_instance = instance.get('brand', '')
         instance_name = instance.get('name', '')
         testing_client = build.servers[0].reconnect_client()
         # If there is a failure, __test_integration_instance will print it
-        success, _ = __test_integration_instance(testing_client, instance)
+        if integration_of_instance not in build.unmockable_integrations and use_mock:
+            logging.debug(f'Integration {integration_of_instance} is mockable, running test-module with mitmproxy')
+            has_mock_file = build.proxy.has_mock_file(integration_of_instance)
+            success = False
+            if has_mock_file:
+                with run_with_mock(build.proxy, integration_of_instance) as result_holder:
+                    success, _ = __test_integration_instance(testing_client, instance)
+                    result_holder[RESULT] = success
+            if not success:
+                logging.debug(f'Playback mode for integration {integration_of_instance} has failed, recording.')
+                with run_with_mock(build.proxy, integration_of_instance, record=True) as result_holder:
+                    success, _ = __test_integration_instance(testing_client, instance)
+                    result_holder[RESULT] = success
+                    logging.debug(f'Record mode for integration {integration_of_instance} has failed.')
```

Comment:
You haven't actually checked the result of the `record` run, so why are you logging that it failed?